BaragonAgentService/src/main/java/com/hubspot/baragon/agent/lbs/FilesystemConfigHelper.java

```diff
@@ -63,13 +63,22 @@ public void checkAndReload() throws InvalidConfigException, LbAdapterExecuteExce
     final BaragonService service = context.getService();
     final boolean previousConfigsExist = configsExist(service);
     Collection<BaragonConfigFile> newConfigs = configGenerator.generateConfigsForProject(context);
-    if (previousConfigsExist && newConfigs.equals(readConfigs(service))) {
+    if (previousConfigsExist && configsMatch(newConfigs, readConfigs(service))) {
       return Optional.absent();
     } else {
       return Optional.of(newConfigs);
     }
   }
 
+  public boolean configsMatch(Collection<BaragonConfigFile> newConfigs, Collection<BaragonConfigFile> currentConfigs) {
+    for (BaragonConfigFile file : newConfigs) {
+      if (!currentConfigs.contains(file)) {
+        return false;
+      }
+    }
+    return true;
```

Comment:
The purpose of the check is to determine if the specified config in newConfigs needs to be applied or if it already exists in the filesystem (currentConfigs). Currently we don't remove extra configs that might be in the filesystem so the check should be fine